### PR TITLE
Prep for 0.1.12-rc2 release

### PR DESCRIPTION
**Pull request by sanderpick**
- More control over IPNS pub and resolve
- Limits sqlite db connections
- Updates IPFS to 0.4.20
- Command-line tweaks for "a tour of textile"

### CODE REVIEW

File: cmd/threads.go [hunk 1]
```diff
@@ -71,8 +71,9 @@ type addThreadsCmd struct {
 	Member     []string       `short:"m" long:"member" description:"A contact address. When supplied, the thread will not allow additional peers, useful for 1-1 chat/file sharing. Can be used multiple times to include multiple contacts.'"`
 	Schema     string         `long:"schema" description:"Thread schema ID. Supersedes schema filename."`
 	SchemaFile flags.Filename `long:"schema-file" description:"Thread schema filename. Supersedes the built-in schema flags."`
-	CameraRoll bool           `long:"camera-roll" description:"Use the built-in camera roll Schema."`
-	Media      bool           `long:"media" description:"Use the built-in media Schema."`
+	Blob       bool           `long:"blob" description:"Use the built-in blob schema for generic data."`
```
Comment (sanderpick) [inline cmd/threads.go]: Adds `--blob` flag.

File: cmd/feed.go [hunk 2]
```diff
@@ -10,28 +10,28 @@ import (
 )
 
 func init() {
-	register(&lsCmd{})
+	register(&feedCmd{})
 }
 
-type lsCmd struct {
+type feedCmd struct {
```
Comment (sanderpick) [inline cmd/feed.go]: Replaced `ls` command with `feed`.

File: bin/collect-profiles.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
```
Comment (sanderpick) [inline bin/collect-profiles.sh]: Script to collect profiles at runtime taken from go-ipfs.

File: core/api.go [hunk 4]
```diff
@@ -283,6 +283,21 @@ func (a *api) Start() {
 			conf.PATCH("", a.patchConfig)
 		}
 	}
+
+	debug := router.Group("/debug/pprof/")
```
Comment (sanderpick) [inline core/api.go]: Adds pprof debug endpoints.

File: repo/db/db.go [hunk 5]
```diff
@@ -41,18 +42,22 @@ type SQLiteDatastore struct {
...
-	var dbPath string
-	dbPath = path.Join(repoPath, "datastore", "mainnet.db")
+	dbPath := path.Join(repoPath, "datastore", "mainnet.db")
 	conn, err := sql.Open("sqlite3", dbPath)
 	if err != nil {
 		return nil, err
 	}
+	conn.SetConnMaxLifetime(time.Minute * 30)
+	conn.SetMaxIdleConns(2)
+	conn.SetMaxOpenConns(4)
```
Comment (sanderpick) [inline repo/db/db.go]: Sets connection limits.

File: textile.go [hunk 6]
```diff
@@ -162,6 +172,58 @@ func (x *commandsCmd) Execute(args []string) error {
 	return nil
 }
 
+func (x *docsCmd) Execute(args []string) error {
```
Comment (sanderpick) [inline textile.go]: New command `docs` generates a giant markdown file with all the commands usages text. Sub commands are further indented.

File: core/cafe_service.go [hunk 7]
```diff
@@ -393,9 +393,51 @@ func (h *CafeService) notifyClient(pid peer.ID) error {
 }
 
 // sendCafeRequest sends an authenticated request, retrying once after a session refresh
-func (h *CafeService) sendCafeRequest(
-	cafe peer.ID, envFactory func(*pb.CafeSession) (*pb.Envelope, error)) (*pb.Envelope, error) {
+func (h *CafeService) sendCafeRequest(cafe peer.ID, envFactory func(*pb.CafeSession) (*pb.Envelope, error)) (*pb.Envelope, error) {
```
Comment (sanderpick) [inline core/cafe_service.go]: This change breaks out HTTP based cafe requests from p2p based requests, which was/is useful for debugging (led me to the db connection limit issue):
- `service.sendCafeRequest`
- `service.sendCafeHTTPRequest`

**Comment by asutula** [comment]
Just scanning for any changes to the mobile API... I don't see any, but let me know if I missed anything.

**Comment by sanderpick** [comment]
Correct, don't think there are any

**Pull request merged**